### PR TITLE
[WIP] Integrate debugger stepping (#212)

### PR DESCRIPTION
**Pull request by Jacoby6000**
Addresses #212

There's still some more debugger stuff that can be done, like updating values, and probably highlighting the line currently suspended.  There could also probably be something we can do for showing where existing breakpoints are, and maybe listing symbols in a quickfix menu or something.

I'm still new to vim plugin development, so I'd like help figuring those parts out!

Please let me know if there's anything I need to change.

### CODE REVIEW

**Comment by jvican** [comment]
It's great, @Jacoby6000. I like the shape of it, and I hope to see the missing features in another PR :wink:. LGTM.

File: ensime_shared/debugger.py [hunk 1]
```diff
@@ -54,6 +54,31 @@ def debug_continue(self, args, range=None):
...
+            "typehint":"DebugStepOutReq",
+            "threadId":self.debug_thread_id})
+ 
+    def debug_next(self, args, range=None):
+        self.log("debug_next: in") 
+        self.send_request({
+            "typehint":"DebugNextReq",
+            "threadId":self.debug_thread_id})
+
+    def debug_locate_name(self, args, range=None):
```
Comment (ChrisCoffey) [inline ensime_shared/debugger.py]: This returns a `DebugLocation`, which is unhandled right now. Please implement a handler for this function or add `debug_locate_name` in a  separate PR.

**Comment by ChrisCoffey** [comment]
Other than the `DebugLocation` comment this looks great.

**Comment by Jacoby6000** [comment]
I'm going to remove `DebugLocation` for this PR.  I'd like for this to get in sooner rather than later, and I don't have time at the moment to finish integrating that functionality.

**Comment by ChrisCoffey** [comment]
That sounds fine to me. We'll get that added soon though :)

**Comment by ches** [comment]
@Jacoby6000 Since #287 got merged, would you mind rebasing and adding documentation for the feature to `doc/ensime.txt` on this branch? There are already sections for debugger commands and debugger usage guide.

**Comment by Jacoby6000** [comment]
@ches I added some documentation to the things that I brought in.. Not sure I provided the best descriptions though.  I don't really know how to define step-out / step-over.

File: ensime_shared/ensime.py [hunk 2]
```diff
@@ -552,12 +552,7 @@ def handle_doc_uri(self, call_id, payload):
         if not self.en_format_source_id:
             self.log("handle_string_response: received doc path")
             port = self.ensime.http_port()
-            
-            url = payload["text"]
-            
-            if not url.startswith("http"):
```
Comment (ChrisCoffey) [inline ensime_shared/ensime.py]: Is there a reason you removed this code?

**Comment by ChrisCoffey** [comment]
This looks solid, aside from a small comment about some deleted code

File: ensime_shared/ensime.py (hunk 2 above)
Comment (ktonga) [inline ensime_shared/ensime.py]: He, unintentionally,  pushed a commit intended for another PR in this one. So this removed lines are the product of reverting that commit.
Now that #283 is merged, I think he should `rebase -i` to squash all commit but the revert one which should be discarded.

**Comment by ChrisCoffey** [comment]
@Jacoby6000 If you can rebase & add back the non-local documentation fix this is good to land

**Comment by jvican** [comment]
@Jacoby6000 Could you rebase please :smile: ? I'll merge it asap.

**Comment by jvican** [comment]
ping @Jacoby6000

**Comment by ChrisCoffey** [comment]
@Jacoby6000 I'd like to land this, so if you're unable to address it please let me know and I'll be happy to clean it up and merge for you.

**Comment by ktonga** [comment]
Closing in favor of #331

**Pull request closed without merging**